wrangler-api/src/main/java/io/cdap/wrangler/api/Row.java

```diff
@@ -150,8 +168,11 @@ public Row add(String name, Object value) {
    * @param idx for which the value and column are removed.
    */
   public Row remove(int idx) {
-    columns.remove(idx);
+    String name = columns.remove(idx);
     values.remove(idx);
+    if (name != null && nameIndex != null) {
+      nameIndex.remove(name.toLowerCase(Locale.ENGLISH));
```

Comment:
remove from the middle will also change the index after it , right ?

Reply:
Yes, good catch, fixed.